topdown/http.go

```diff
@@ -68,41 +69,80 @@ type httpSendKey string
 const httpSendBuiltinCacheKey httpSendKey = "HTTP_SEND_CACHE_KEY"
 
 func builtinHTTPSend(bctx BuiltinContext, args []*ast.Term, iter func(*ast.Term) error) error {
-
-	bctx.Metrics.Timer(httpSendLatencyMetricKey).Start()
-
 	req, err := validateHTTPRequestOperand(args[0], 1)
 	if err != nil {
 		return handleBuiltinErr(ast.HTTPSend.Name, bctx.Location, err)
 	}
 
-	reqExecutor, err := newHTTPRequestExecutor(bctx, req)
+	raiseError, err := getRaiseErrorValue(req)
+	if err != nil {
+		return handleBuiltinErr(ast.HTTPSend.Name, bctx.Location, err)
+	}
+
+	return builtinHTTPSendHelper(bctx, req, raiseError, iter)
+}
+
+func builtinHTTPSendHelper(bctx BuiltinContext, req ast.Object, raiseError bool, iter func(*ast.Term) error) (err error) {
```

Comment:
I think this code can be simplified quite a bit: https://gist.github.com/tsandall/deeaff535f9eaae5a55559067611473e

There are a couple red flags:

* the `httpSendErrType` variable is a bit brittle. E.g., if we add another error path in the future, we need to make sure to set the variable accordingly; if we forget, the built-in won't behave correctly.

* the defer block is re-executing the iterator; this means the search after this expression will be re-evaluated unnecessarily

* generally, appending "helper" to a function name is an indicator that the function is doing too much--in this case, the solution is to split the iterator invocation from the http request execution--that avoids the need for the defer and the boolean.

Reply:
Refactored the code as per your comment.